sdk-ts/src/mint/token.ts

```diff
@@ -0,0 +1,183 @@
+import { PublicKey, TransactionInstruction } from "@solana/web3.js";
+import { TokenInfo } from "@solana/spl-token-registry";
+import { TOKEN_PROGRAM_ID } from "@solana/spl-token";
+import * as anchor from "@project-serum/anchor";
+
+import { getOrCreateATAInstruction, toDecimal } from "../utils";
+
+import { Strategy } from "./index";
+import { StrategyHandler, StrategyType } from "../strategy";
+import NativeToken from "./native";
+import { getVaultPdas } from "../utils";
+import Vault from "../vault";
+
+export default class MintToken extends NativeToken {
+  constructor(
+    tokenInfo: TokenInfo,
+    strategyProvider: Map<StrategyType, StrategyHandler>,
+    vault: Vault
+  ) {
+    super(tokenInfo, strategyProvider, vault);
+  }  
```

Comment:
I don't think we should be extending from NativeToken, since most of the function implementation are redefined here anyway, extending will just create more confusion